gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -185,42 +171,28 @@ public Object call() throws IOException, InterruptedException {
 
     private Object doResumableUpload() throws IOException, InterruptedException {
       // Send the initial StartResumableWrite request to get an uploadId.
-      String uploadId = getUploadId();
-      // Protect against chunkSize changing mid RPC.
-      int chunkSize = uploadChunkSize;
+      String uploadId = startResumableUpload();
       InsertChunkResponseObserver responseObserver;
-      // This ByteString holds all the data for the current streaming RPC that hasn't
-      // yet been confirmed committed. This should never be larger than `chunkSize`
-      // defined above.
-      ByteString chunkData = ByteString.EMPTY;
+
       long writeOffset = 0;
       Hasher objectHasher = Hashing.crc32c().newHasher();
       do {
+        ByteString chunkData = ByteString.EMPTY;
         responseObserver =
-            new InsertChunkResponseObserver(
-                uploadId, chunkData, chunkSize, writeOffset, objectHasher);
+            new InsertChunkResponseObserver(uploadId, chunkData, writeOffset, objectHasher);
         stub.insertObject(responseObserver);
         responseObserver.done.await();
-
-        chunkData = responseObserver.getChunkData();
-        // We can assume all bytes were written on a successful final request.
-        long bytesCommitted =
-            responseObserver.isFinished()
-                ? writeOffset + chunkData.size()
-                : getCommittedWriteSize(uploadId);
-
-        // This should never fail since uploadChunkSize is an int.
-        int chunkBytesCommitted = Math.toIntExact(bytesCommitted - writeOffset);
-        if (checksumsEnabled && !responseObserver.isFinished()) {
-          for (ByteBuffer buffer : chunkData.asReadOnlyByteBufferList()) {
-            objectHasher.putBytes(buffer);
-          }
-        }
-        chunkData = chunkData.substring(chunkBytesCommitted);
-        writeOffset = bytesCommitted;
+        writeOffset += responseObserver.bytesWritten();
       } while (!responseObserver.isFinished());
 
       if (responseObserver.hasError()) {
+        // TODO(b/150892988): Support resuming an upload after a transient error as follows:
+        //  1. Build a wrapper class around the ByteString (or subclass it), and keep track of the
+        //     start offset of each chunk.
+        //  2. Build a list of chunks, and (a) use it for rewinding when restarting at last
+        //     committed offset (per call to getCommittedWriteSize); (b) remove chunks from the list
+        //     that have been persisted.
+        //  3. Limit the list size to some number (possibly flag-controlled) of sent chunks.
         logger.atSevere().withCause(responseObserver.getError()).log("Error:");
```

Comment:
Remove this log statement because this is an anti-pattern - we need to either throw or log, but not both.

Reply:
Updated in my forked repo (https://github.com/mfschwartz/hadoop-connectors/blob/grpc-2.0.0/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java). Should I generate a new pull request with that update, or how do you like to do this?

Reply:
No, when you push this new commit to the `grpc-2.0.0` branch in you repo it automatically added to this PR.